File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        
        int sum = 0,ans = nums.size() + 1,start = 0;
        
        
        for(int i=0;i<nums.size();i++){
            
            sum+=nums[i];
            
            while(sum>=target){
                ans = min(ans,i - start +1);
                sum-=nums[start++];
                
            }
            
        }
        
        
        if(ans == nums.size() + 1)
            return 0;
        
        
        return ans;
        
    }
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py

        

class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        currSum=0;
        start = 0;
        ans = len(nums) + 1;
        
        
        for i in range(0,len(nums)) :
            currSum = currSum + nums[i];
            
            while(currSum>=target) :
                ans = min(ans,i - start+1);
                currSum = currSum - nums[start];
                start = start + 1;
                
        
        return 0 if(ans == len(nums) + 1) else ans;
        